deepchecks/utils/correlation_methods.py

```diff
@@ -0,0 +1,111 @@
+
+        Wikipedia: https://en.wikipedia.org/wiki/Conditional_entropy
+        Parameters:
+        -----------
+        x: Union[List, np.ndarray, pd.Series]
+            A sequence of numerical_feature without nulls
+        y: Union[List, np.ndarray, pd.Series]
+            A sequence of numerical_feature without nulls
+        Returns:
+        --------
+        float
+            Representing the conditional entropy
+    """
+    y_counter = Counter(y)
+    xy_counter = Counter(list(zip(x, y)))
+    total_occurrences = sum(y_counter.values())
+    entropy = 0.0
+    for xy in xy_counter.keys():
+        p_xy = xy_counter[xy] / total_occurrences
+        p_y = y_counter[xy[1]] / total_occurrences
+        entropy += p_xy * math.log(p_y / p_xy, math.e)
+    return entropy
+
+
+def theil_u_correlation(x: Union[List, np.ndarray, pd.Series], y: Union[List, np.ndarray, pd.Series]) -> float:
+    """
+        Calculates the Theil's U correlation of y to x.
+
+        Theil's U is an asymmetric measure ranges [0,1] based on entropy which answers the question: how well does
+        feature y explains feature x? For more information see https://en.wikipedia.org/wiki/Uncertainty_coefficient
+        Parameters:
+        -----------
+        x: Union[List, np.ndarray, pd.Series]
+            A sequence of a categorical feature values without nulls
+        y: Union[List, np.ndarray, pd.Series]
+            A sequence of a categorical feature values without nulls
+        Returns:
+        --------
+        float
+            Representing the theil_u correlation between y and x
+    """
+    s_xy = conditional_entropy(x, y)
+    x_values_counter = Counter(x)
+    total_occurrences = sum(x_values_counter.values())
+    values_probabilities = list(map(lambda n: n / total_occurrences, x_values_counter.values()))
+    s_x = entropy(values_probabilities)
+    if s_x == 0:
+        return 1
+    else:
+        return (s_x - s_xy) / s_x
+
+
+def correlation_ratio(categorical_feature: Union[List, np.ndarray, pd.Series],
+                      numerical_feature: Union[List, np.ndarray, pd.Series]) -> float:
+    """
+        Calculates the correlation ratio of numerical_feature to categorical_feature.
+
+        Correlation ratio is an asymmetric variance based method which answers the question: how well does a
+        numeric feature explains a categorical feature? returns a value in [0,1].
+        For more information see https://en.wikipedia.org/wiki/Uncertainty_coefficient
+        Parameters:
+        -----------
+        categorical_feature: Union[List, np.ndarray, pd.Series]
+            A sequence of a categorical feature values without nulls
+        numerical_feature: Union[List, np.ndarray, pd.Series]
+            A sequence of a numerical feature values without nulls
+        Returns:
+        --------
+        float
+            Representing the correlation ratio between the features.
+    """
```

Comment:
If you require values without nulls, why not do a null validation?

Reply:
It is computationally better to it once for the all dataset in the check